packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit_webview_widget.dart

```diff
@@ -451,11 +483,67 @@ class WebKitWebViewPlatformController extends WebViewPlatformController {
         },
       );
     } else {
-      webView.observeValue = null;
+      webView.setObserveValue(null);
       return webView.removeObserver(webView, keyPath: 'estimatedProgress');
     }
   }
 
+  void _setJavaScriptMode(JavascriptMode mode) {
+    switch (mode) {
+      case JavascriptMode.disabled:
+        webView.configuration.preferences.setJavaScriptEnabled(false);
+        break;
+      case JavascriptMode.unrestricted:
+        webView.configuration.preferences.setJavaScriptEnabled(true);
+        break;
+    }
+  }
+
+  void _setUserAgent(WebSetting<String?> userAgent) {
+    if (userAgent.isPresent) {
+      webView.setCustomUserAgent(userAgent.value);
+    }
+  }
+
+  Future<void> _setZoomEnabled(bool zoomEnabled) async {
+    if (_zoomEnabled == zoomEnabled) {
+      return;
+    }
+
+    _zoomEnabled = zoomEnabled;
+    if (!zoomEnabled) {
+      await _disableZoom();
+    } else {
+      // WkWebView does not support removing a single user script, so instead we
+      // remove all user scripts, all message handlers, and re-register channels
+      // that shouldn't be removed. Note that this workaround could interfere
+      // with exposing support for custom scripts from applications.
+      webView.configuration.userContentController.removeAllUserScripts();
+      webView.configuration.userContentController
+          .removeAllScriptMessageHandlers();
+
+      final Set<String> javaScriptChannelNames =
+          _scriptMessageHandlers.keys.toSet();
+      _scriptMessageHandlers.clear();
+
+      await addJavascriptChannels(javaScriptChannelNames);
```

Comment:
This is a non-trivial block of code that's being duplicated; it should be extracted to a helper that re-syncs to a desired state (current scripts/handlers, zoom state).

Reply:
Thats a good point. Added `_resetUserScripts` and added tests.